### DISCUSSION TITLE
BUG: Incorrect Creation/Modification Date for Uploaded Files

### DISCUSSION THREAD

**Discussion by jtamm-red** [photoprism/photoprism#4043] [Q&A]
#### 1. What is not working as documented?
The uploaded file displays an incorrect date, reflecting the date of upload rather than the original file's date. It is crucial for the file date to accurately reflect the original creation date of the file. PhotoPrism relies on extracting EXIF metadata exclusively from the file itself to determine the correct date. Therefore, it is imperative to rectify this discrepancy to ensure accurate and meaningful organization of files within the system.

#### 2. How can we reproduce it?

1. Launch PhotoSync on your Android device.
2. Tap the red sync icon.
3. Navigate to the 'Send' tab and select 'New.'
4. Choose 'PhotoPrism' as the destination.
Please note that the transfer process may take a few minutes.

To verify the uploaded date of the transferred files on the PhotoPrism server:
1. Access the PhotoPrism server terminal.
`docker exec -it photoprism bash`
2. Navigate to the directory containing the original files.
`cd /photoprism/originals/`
3. List all folders in the current directory.
`ls -la`
4. Enter the sub-folder where your files are located.
`cd Camera`
5. List all files in the Camera.
`ls -la`
5. To check the upload date of a specific file, run the following command:
`stat IMG_20230412_133303.jpg`
This will provide detailed information, including the file's upload date.
You can make a comparison between the file's birth date and its filename, specifically looking for the datetime '20230412_133303' (which corresponds to April 12, 2023, 13:33 GMT+3).

#### 3. What behavior do you expect?

The file's date should accurately reflect the original creation date. This is crucial for easy sorting by date in file explorers. For instance, during maintenance tasks such as copying files to a backup external USB HDD, it is imperative that the file dates are not incorrect. Maintaining accurate file dates ensures a smooth and efficient organization of files, facilitating hassle-free processes such as backups.

#### 4. What could be the cause of your problem?

Dealing with files that have incorrect dates can lead to confusion and hinder the ability to filter or sort them based on date. In such cases, relying on EXIF metadata becomes necessary, but reading this metadata can be time-consuming, especially when dealing with a large number of files, such as 100,000. Unfortunately, standard file explorers lack built-in EXIF reading features, requiring the use of external applications. This process can be cumbersome and complex, making it challenging to efficiently manage and organize files with accurate date information.

Sometimes the date appears as unknown on the PhotoPrism page (refer to the screenshot below). I have to manually inspect each file and update the correct date, a process that consumes a significant amount of time.

#### 5. Can you provide us with example files for testing, error logs, or screenshots?

The datetime display on the PhotoPrism page appears unusual, with the time consistently showing as '00:00:00'.
![image](https://github.com/photoprism/photoprism/assets/53046846/ed5fa957-3267-49b7-bfda-923a6575de19)

Compare the date of upload with the original creation date of the file.
![image](https://github.com/photoprism/photoprism/assets/53046846/66e0a26a-6640-43b9-ab2b-cfed4a8eda86)

The file date is accurately displayed on the Android Explorer.
![image](https://github.com/photoprism/photoprism/assets/53046846/61e2251e-f570-42f0-b1d6-68dd466bd93a)

Curiously, the fields for day, month, and year appear as 'Unknown,' while the time is displayed correctly.
![image](https://github.com/photoprism/photoprism/assets/53046846/32129b8e-dc2b-4dd8-9576-36b713fdd2d8)

Upon executing the command 'stat 20240113_192341_635.mp4,' it is observed that the Birth and Change dates are displayed incorrectly. The timestamp '2024-01-26 21:31:15' signifies the time at which the file was uploaded.
![image](https://github.com/photoprism/photoprism/assets/53046846/8a5c386e-8b6e-4793-aa2f-9dde6dfccab0)

The datetime is accurately displayed on the Android Explorer.
![image](https://github.com/photoprism/photoprism/assets/53046846/7240b507-d95f-4b22-903c-7bc234ae6bd1)

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: ARM64 231128-f48ff16ef

(b) Database Type & Version: MariaDB 11.1.2+maria~ubu2204

(c) Operating System Types & Versions: Linux 5.10.160

(d) Docker 20.10.22, build 3a2c30b

(e) PhotoSync 4.0.10 (5786) for android

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: NanoPi R6C / RK3588S

(b) Physical Memory: 8GB

(c) Storage Type: 32GB eMMC, 500GB NVMe

#### 8. Do you use a Reverse Proxy, Firewall, VPN, or CDN?

No

**Comment by graciousgrey** [comment]
Since "Unknown" is displayed in the editing dialog, I assume that your photos only have a modification date, but no real taken date.

If there is no taken at date we do use the file modification date for sorting, but we do not set it as the date taken because it is not necessarily the date the photo was taken.

You can see the metadata of your files by running `exiftool -j filename.jpg`

You can find the metadata fields that are used as taken at date in the following overview:

https://www.photoprism.app/kb/metadata

In case of your whats app images we extracted the date from the filename this is why day, month and year are set but the time is 00:00.

Currently, the modification date of files is updated when the files are synchronized with webdav. Improvements for this behavior are already in progress.

- https://github.com/photoprism/photoprism/issues/3959
- https://github.com/photoprism/photoprism/issues/3942

**Reply by jtamm-red** [reply]
Thanks for quick response!

I tried to execute command `exiftool 20240113_192341_635.mp4`

```
ExifTool Version Number         : 12.65
File Name                       : 20240113_192341_635.mp4
Directory                       : .
File Size                       : 100 MB
File Modification Date/Time     : 2024:01:26 21:31:26+00:00
File Access Date/Time           : 2024:01:31 11:39:18+00:00
File Inode Change Date/Time     : 2024:01:26 21:31:26+00:00
File Permissions                : -rw-rw-r--
File Type                       : MP4
File Type Extension             : mp4
MIME Type                       : video/mp4
Major Brand                     : MP4 Base Media v1 [IS0 14496-12:2003]
Minor Version                   : 0.2.0
Compatible Brands               : isom, iso2, avc1, mp41
Media Data Size                 : 99643761
Media Data Offset               : 48
Movie Header Version            : 0
Create Date                     : 0000:00:00 00:00:00
Modify Date                     : 0000:00:00 00:00:00
Time Scale                      : 1000000
Duration                        : 0:00:30
Preferred Rate                  : 1
Preferred Volume                : 100.00%
Preview Time                    : 0 s
Preview Duration                : 0 s
Poster Time                     : 0 s
Selection Time                  : 0 s
Selection Duration              : 0 s
Current Time                    : 0 s
Next Track ID                   : 3
Track Header Version            : 0
Track Create Date               : 0000:00:00 00:00:00
Track Modify Date               : 0000:00:00 00:00:00
Track ID                        : 1
Track Duration                  : 0:00:30
Track Layer                     : 0
Track Volume                    : 0.00%
Image Width                     : 1080
Image Height                    : 1920
Graphics Mode                   : srcCopy
Op Color                        : 0 0 0
Compressor ID                   : avc1
Source Image Width              : 1080
Source Image Height             : 1920
X Resolution                    : 72
Y Resolution                    : 72
Bit Depth                       : 24
Color Profiles                  : nclx
Color Primaries                 : BT.709
Transfer Characteristics        : BT.709
Matrix Coefficients             : BT.709
Buffer Size                     : 0
Max Bitrate                     : 26241126
Average Bitrate                 : 26241126
Video Frame Rate                : 29.97
Matrix Structure                : 1 0 0 0 1 0 0 0 1
Media Header Version            : 0
Media Create Date               : 0000:00:00 00:00:00
Media Modify Date               : 0000:00:00 00:00:00
Media Time Scale                : 48000
Media Duration                  : 29.78 s
Media Language Code             : und
Handler Description             : SoundHandler
Balance                         : 0
Audio Format                    : mp4a
Audio Channels                  : 2
Audio Bits Per Sample           : 16
Audio Sample Rate               : 48000
Make                            : Insta360
Model                           : X3.VIDEO_NORMAL
Handler Type                    : Metadata
Handler Vendor ID               : Apple
Encoder                         : Lavf60.3.100
Image Size                      : 1080x1920
Megapixels                      : 2.1
Avg Bitrate                     : 26.4 Mbps
Rotation                        : 0
```

I observe a 'Create Date' value of '0000:00:00 00:00:00,'.
I propose that PhotoPrism should take the file's datetime if value (Create Date) is empty in the EXIF data.

**Reply by lastzero** [reply]
@jtamm-red As mentioned in the issues referenced above, the `photoprism/photoprism:test` image available on Docker Hub includes support for setting the mtime based on the X-OC-MTime header:
- https://github.com/photoprism/photoprism/issues/3942#issuecomment-1881575630

Note that this is a custom, non-standard header and therefore only works if the client - in your case PhotoSync - sets this header. We would therefore appreciate it if you could help us test these changes and provide feedback on them (by using the test image) before we make it available to everyone in the upcoming release.

**Reply by lastzero** [reply]
I've also added a note to the other (related) issue so anyone interested in this feature can easily find it to help with testing:
- https://github.com/photoprism/photoprism/issues/3959#issuecomment-1919866131

**Reply by lastzero** [reply]
If the WebDAV client/app you are using does not submit the file modification date, there is not much we can do. However, it should be easy for app developers to add it if you ask them to. Thank you!
